### PR TITLE
Implemented Distributed Data Parallel(DDP)

### PR DESCRIPTION
**Pull request by chelseajohn**
This PR implements [PyTorch DDP](https://pytorch.org/docs/stable/notes/ddp.html) for training and inference.

Training is done by replicating the model on different GPUs and synchronizing the gradients. A simple schema can be found in `docs/DDP.svg`

For inference, `nSamples` is divided among the GPU's such that there is only one pass through `DataLoader` for now (sample per GPU = nSamples/number of GPUs), after `nEvals` is done the output of all GPU's is concatenated to form the whole output.

Example submission scripts for training and evaluation of model with DDP can be found at `utils/launch_scripts/submit_ddp_training.sbatch.sh` and  `utils/launch_scripts/submit_eval.sbatch.sh`

It is possible to do DDP inference with models trained with or without using DDP.

### CODE REVIEW

**Comment by tlunet** [comment]
So, if I understood correctly, you parallelized over the samples, right ?

**Comment by sgoetschel** [comment]
Can you maybe add a readme somewhere explaining how to use it and what is to be considered when training/evaluating in data-parallel (or training in parallel + evaluation in serial)? Is it possible to add some tests to make sure nothing breaks?

**Comment by chelseajohn** [comment]
I have added a section for DDP in `docs/README.md` and made a test for DDP training.

**Review by sgoetschel — COMMENTED** [review]
Thanks for the documentation :)
